src/Logic.hs

```diff
@@ -94,7 +94,7 @@ type PushWithTagResult = (Either Text TagName, PushResult)
 
 -- | Error returned when 'TryIntegrate' fails.
 -- It contains the name of the target branch that the PR was supposed to be integrated into.
-newtype IntegrationFailure = IntegrationFailure Branch
+newtype IntegrationFailure = IntegrationFailure (Branch, ReasonToFail)
```

Comment:
I think it'd be nicer to turn this into a regular `data` type and have multiple arguments to the constructor, instead of using a tuple, i.e.

```haskell
data IntegrationFailure = IntegrationFailure Branch ReasonToFail
```